core/src/main/java/org/infinispan/interceptors/locking/ClusteringDependentLogic.java

```diff
@@ -349,6 +360,12 @@ protected void commitSingleEntry(CacheEntry entry, Metadata metadata, FlagAffect
          // TODO: Can the reset be done after notification instead?
          boolean created = entry.isCreated();
          boolean removed = entry.isRemoved();
+         boolean expired;
+         if (entry instanceof MVCCEntry) {
```

Comment:
Not sure this has any measurable performance impact, but could we do the `instanceof` check only if `!created && !removed`?